product_set/models/product_set_line.py

```diff
@@ -0,0 +1,39 @@
+# Copyright 2015 Anybox S.A.S
+# Copyright 2016-2018 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+from odoo import fields, models
+
+
+class ProductSetLine(models.Model):
+    _name = "product.set.line"
+    _description = "Product set line"
+    _rec_name = "product_id"
+    _order = "sequence"
+
+    product_id = fields.Many2one(
+        comodel_name="product.product",
+        domain=[("sale_ok", "=", True)],
+        string="Product",
+        required=True,
+    )
+    quantity = fields.Float(
+        digits="Product Unit of Measure", required=True, default=1.0
+    )
+    product_set_id = fields.Many2one("product.set", string="Set", ondelete="cascade")
+    active = fields.Boolean(
+        string="Active", related="product_set_id.active", store=True, readonly=True
+    )
+    sequence = fields.Integer(required=True, default=0)
+    discount = fields.Float(string="Discount (%)", digits="Discount", default=0.0)
+
+    def prepare_sale_order_line_values(self, order, quantity, max_sequence=0):
```

Comment:
This one should go in `sale_product_set`.

Reply:
What about this?

Reply:
you are right, did not see it. -> Done, but I've left the sequence in the product_set part.